python/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# =============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.20.1 FATAL_ERROR)
+
+# TODO: Pull this directly from the file written by rapids_cmake_write_version_file.
+set(rmm_version 22.04.00)
+
+project(
+  rmm-python
+  VERSION ${rmm_version}
+  # TODO: Building Python extension modules via the python_extension_module requires the C language
+  # to be enabled here. The test project that is built in scikit-build to verify various linking
+  # options for the python library is hardcoded to build with C, so until that is fixed we need to
+  # keep this.
+  LANGUAGES C CXX CUDA)
```

Comment:
Do we need the `CUDA` language here? Isn't it all host code?

Reply:
I believe we don't even use the `CUDA` language in the RMM C++ project so I can't imagine it being an issue here.

https://github.com/rapidsai/rmm/blob/branch-22.04/CMakeLists.txt#L29

Reply:
I was able to remove the `find_package` for CUDAToolkit because that is correctly being pulled in from adding the rmm include directory for the C++ build if needed. However, it seems that because we are linking to rmm with `target_link_libraries(${cython_module} rmm::rmm)` we also need to include the languages required by that link. If I remove the CUDA language I get a CMake error `Cannot use features from non-enabled language CUDA`. If I comment out that link line then the configuration succeeds (of course the build fails because it can't find RMM headers).

I find it interesting that the configuration for the C++ part of the project doesn't list CUDA as a language at all though. However, it does link to `CUDA::*` targets as interface libraries. Perhaps that is why the dependency is being transmitted to the Python build even though the C++ config doesn't require CUDA? @robertmaynard could you shed any light on what might be going on here?

Reply:
Ahh I think I see it. Since RMM is header-only, it doesn't need CUDA *unless you build tests or benchmarks*. And indeed, we see that the language is conditionally enabled here: https://github.com/rapidsai/rmm/blob/branch-22.04/CMakeLists.txt#L84.

Reply:
My best guess is it's coming from the build interface definition here: https://github.com/rapidsai/rmm/blob/branch-22.04/CMakeLists.txt#L79

I think that's being set so any CUDA code that includes RMM inherits the C++17 requirement, but in our case here it forces the need for CUDA language unnecessarily.